### PR TITLE
Makes construction graphs turing complete

### PR DESCRIPTION
**Pull request by Zumorica**
## About the PR
Construction graphs now have all tools necessary to theoretically compute any algorithm ever.
Not like you would ever want to do that, however...
I have included one example construction graph, `turing.yml`, which implements a 2-bit adder.

**Screenshots**
Peek 2021-03-05 06-54

## Why.
I don't know, it's 6 AM and I haven't slept yet... And construction graphs are suddenly turing complete. Yet another buzzword to add to the mix.

### CODE REVIEW

File: Resources/Prototypes/Recipes/Construction/Graphs/turing.yml [hunk 1]
```diff
@@ -0,0 +1,296 @@
+- type: constructionGraph
```
Comment (PaulRitter) [inline Resources/Prototypes/Recipes/Construction/Graphs/turing.yml:1]: is this supposed to be merged? as an example?

**Review by PaulRitter — APPROVED** [review]
looks good. i just have one question: why

File: Resources/Prototypes/Recipes/Construction/Graphs/turing.yml (hunk 1 above)
Comment (PaulRitter) [reply Resources/Prototypes/Recipes/Construction/Graphs/turing.yml:1]: wait i dont think i fully understood this pr

**Comment by Zumorica** [comment]
> i just have one question: why

We do what we must because we can.

**Comment by moonheart08** [comment]
Are you sure this is TC and not just a pushdown automata? It seems like a pushdown automata to me.

**Comment by Zumorica** [comment]
> Are you sure this is TC and not just a pushdown automata? It seems like a pushdown automata to me.

Gonna state this here too, construction graphs ARE turing complete now. They can pretty much act like Minsky machines, which are proven to be turing complete.
Here's some of the things construction graphs can do now, off the top of my head:
- Have infinite registers (using containers)
- Set or Clear any register
- Do any boolean logic using values on registers
- Move registers around
- Take input from users and store it on registers
- Conditionally go to other states depending on values on registers
- etc

**Comment by tmtmtl30** [comment]
i think this is probably a bad idea
i don't want construction graph shitcode

**Comment by Zumorica** [comment]
> i think this is probably a bad idea
> i don't want construction graph shitcode

construction graph shitcode won't get merged

**Pull request merged**